pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -1029,6 +1030,29 @@ func (dr *DashboardServiceImpl) FindDashboards(ctx context.Context, query *dashb
 		return finalResults, nil
 	}
 
+	// This is the key part which ensures all queries for folders go through folder service.
+	if (query.Type == searchstore.TypeFolder) || (query.Type == searchstore.TypeAlertFolder) {
+		folderQuery := folder.FindPersistedFoldersQuery{
+			Title: query.Title,
+			OrgId: query.OrgId,
+			// #TODO fill in the rest
```

Comment:
It looks like there are still some todo's left? Here & a bit below

Reply:
Thank you! Looks like this change isn't needed anymore (commented below).